classes/blocks/class-media.php

```diff
@@ -23,41 +23,22 @@ class Media extends Base_Block {
 	 */
 	const BLOCK_NAME = 'media_video';
 
-	/**
-	 * Register shortcake shortcode.
-	 *
-	 * @param array  $attributes Shortcode attributes.
-	 * @param string $content   Content.
-	 *
-	 * @return mixed
-	 */
-	public function add_block_shortcode( $attributes, $content ) {
-
-		$attributes = shortcode_atts(
-			[
-				'video_title'      => '',
-				'description'      => '',
-				'youtube_id'       => '',
-				'video_poster_img' => '',
-			],
-			$attributes,
-			'shortcake_media_video'
-		);
-
-		return $this->render( $attributes );
-	}
-
 	/**
 	 * Media constructor.
 	 */
 	public function __construct() {
-		add_shortcode( 'shortcake_media_video', [ $this, 'add_block_shortcode' ] );
-
 		register_block_type(
 			'planet4-blocks/media-video',
 			[
 				'editor_script'   => 'planet4-blocks',
-				'render_callback' => [ $this, 'render' ],
+				'render_callback' => static function ( $attributes, $content ) {
+					if ( '' === trim( $content ) ) {
+						$json = wp_json_encode( [ 'attributes' => $attributes ] );
```

Comment:
Instead of setting up the block to be frontend rendered here, we could also fetch the dynamic data here already. This will not add to the page's response time as the result is cached along with the rest of the page, and will periodically update as the cache gets invalidated, which should be more than sufficient as this data changes infrequently.

Reply:
```
json = wp_json_encode( 
    [ 'attributes' => array_merge($attributes, Media::get_media_data($attributes) ] 
);
```

Reply:
Then the only thing we need to do is remove the `useMediaData` hook call, as the resulting attributes will be there already.
```javascript
export const MediaFrontend = (props) => {
  // const mediaData = useMediaDataFetch(props);

  // const staticProps = { ...props, ...mediaData };

  const staticProps = props;

  return <StaticMediaFrontend { ...staticProps } />
}
```